src/advanced/merkelized-list.md

```diff
@@ -108,58 +108,36 @@ Let `T(height, index)` be a value at tree node for element `index` at height
 `height`. Elements `T(0, index)` contain serialized values of the underlying list
 according to [the Exonum binary serialization spec](../architecture/serialization.md).
 Elements `T(height, index)` for `height > 0` are hashes corresponding the following
-rules. Each node is hashed with the corresponding prefix:
+rules.
 
-| Prefix | Node type |
-|--------|-----------|
-| `0x00` | leaf node |
-| `0x01` | branch node |
+#### Rule 1. Empty tree
 
-The root node is hashed with the `0x02` prefix and with the number of elements
-of the list.
-
-```none
-hash( 0x02 || length || root_hash )
-```
-
-Where `root_hash` is the merkle root of the `ProofListIndex`, `length` is
-8-bytes length of the `ProofListIndex` encoded as little endian.
-
-#### Rule 1. Empty Tree
-
-Hash of an empty tree is defined as follows:
-
-```none
-hash( 0x02 || 0 || Hash::default() )
-```
-
-where `0x02` is the root node prefix for `ProofListIndex`,
-`0` is the length of an empty list, `Hash::default()` is 32 zero bytes.
+Hash of an empty tree is defined as 32 zero bytes.
 
 #### Rule 2. `height=1`
 
-Hash of a value contained in `(height = 0, index)` is defined as
+Hash of a value, contained in `(height = 0, index)`, is defined as:
 
 ```none
-T(1, index) = hash(0x00 || T(0, index)).
+T(1, index) = hash(T(0, index)).
 ```
 
-#### Rule 3. `height > 1`, Two Children
+#### Rule 3. `height > 1`, two children
```

Comment:
Should we revert grammar related comments?

Reply:
Good point, I'll put them these changes back. So far, I've simply reverted to the state of the article at a specific commit, so the grammar changes were also reverted.